libs/category/src/selectors/category.selector.ts

```diff
@@ -151,18 +151,21 @@ export const selectCategoryPageProducts = createSelector(
 
 export const selectCategory = createSelector(
 	selectCategoryEntities,
-	(entities, props) => {
-		return entities[props.id];
-	}
+	(entities, props) =>  entities[props.id]
 );
 
 export const selectProductsByCategory = createSelector(
 	selectCategoryEntities,
 	fromProduct.selectAllProducts,
-	(entities, products, props) => {
-		if (!entities[props.id]) {
-			return null;
-		}
-		return products.filter(product => entities[props.id].productIds.indexOf(product.id) >= 0);
-	}
+	(entities, products, props) => entities[props.id] && entities[props.id].product_ids
+		? products.filter(product => entities[props.id].product_ids.indexOf(product.id) >= 0)
+		: null
+);
+
+export const selectTotalProductsByCategory = createSelector(
+	selectCategoryEntities,
+	fromProduct.selectAllProducts,
+	(entities, products, props) => entities[props.id] && entities[props.id].product_ids
+		? products.filter(product => entities[props.id].product_ids.indexOf(product.id) >= 0).length
+		: null
```

Comment:
So this selector is just getting a count of the products? Is there any way that it can use `selectProductsByCategory` to simplify the logic?

Reply:
oh lol, yea

Reply:
It's a little simpler. Probably more useful for more complex selectors, but definitely the right call.

Reply:
hm yeah not as simple as I envisioned it but that seems to be the best way. One last thing, can you only make one call to the other selector, i.e.
```ts
export const selectTotalProductsByCategory = createSelector(
  selectCategoryEntities,
  fromProduct.selectAllProducts,
  (entities, products, props) => {
    const products = selectProductsByCategory.projector(entities, products, { id: props.id });
    return products ? products.length : null
  }
)
```

Reply:
I think it's better this way, because it's just adding an extra subscription to a selector, whereas the other way adds a copy of all products in the app to memory.

Reply:
actually..hmm. nevermind, it'll just save an observable of the products in state. I'll make the change

Reply:
nevermind again. I think it must save the actual array of products (for the category, not all of the products (I misspoke earlier)), because you can check the length of it. So it must be the actual value and not an observable.

Reply:
hm good point. okay I'm good with this then.